src/pages/profile/index.js

```diff
@@ -0,0 +1,205 @@
+import React, { useState, useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { navigate, Link } from 'gatsby'
+import { useFormik } from 'formik'
+import * as Yup from 'yup'
+import Layout from '../../components/Layout'
+import { getUser } from '../../services/util'
+
+import { getProfile, addProfile } from '../../services/api'
+
+const MyProfile = () => {
+  const { t } = useTranslation()
+  const user = getUser()
+  const [error, setError] = useState(null)
+  const [profile, setProfile] = useState({})
+  const [isSaved, setIsSaved] = useState(false)
+
+  useEffect(() => {
+    if (!user) {
+      navigate('/signin')
+    }
+    const getProfileData = async () => {
+      try {
+        const { data } = await getProfile()
+        if (data) {
+          setProfile(data)
+        }
+        setError(null)
+      } catch (err) {
+        if (err.message.match(/(403|400)/)) {
+          setError('You are not a user')
+        } else setError('Something went wrong')
+      }
+    }
+    getProfileData()
+  }, [])
+
+  /*   useEffect(() => {
+    console.log(profile)
+  }, [profile]) */
+
+  const formik = useFormik({
+    enableReinitialize: true,
+    initialValues: {
+      bio: profile.bio || '',
+      linkedin: profile.linkedin || '',
+      github: profile.github || '',
+      twitter: profile.twitter || '',
+    },
+    validationSchema: Yup.object({
+      bio: Yup.string().min(50, 'should be at least 50'),
+      linkedin: Yup.string().url('Should be a link'),
+      github: Yup.string().url('Should be a link'),
+      twitter: Yup.string().url('Should be a link'),
```

Comment:
Can we have those coming from the messages file instead of hardcoded? 
One way of doing this is to add the key here as the error message 

```
bio: Yup.string().min(50, 'validation.required'),
```

in the message.json file 
```
validation.required: 'This field is required'
```

then when we display we simply use the translation function 
```
{t(formik.errors.bio)}
```

Reply:
I don't understand what you wrote here? what is the problem of using the conventional method?
 `bio: Yup.string().min(50, t('atLeast50')),` for example?

Reply:
Works too, I was just giving an example, anyway is fine as long as it is not hardcoded